modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectContentEventListener.java

```diff
@@ -0,0 +1,198 @@
+
+        this.simpleExecutor = new SimpleExecutor( Executors::newSingleThreadExecutor, "project-node-sync-thread",
+                                                  e -> LOG.error( "Project node sync failed", e ) );
+    }
+
+    @Deactivate
+    public void deactivate()
+    {
+        this.simpleExecutor.shutdownAndAwaitTermination( Duration.ZERO, neverCommenced -> {
+        } );
+    }
+
+    @Override
+    public void onEvent( final Event event )
+    {
+        if ( event != null )
+        {
+            if ( !event.isLocalOrigin() )
+            {
+                return;
+            }
+
+            if ( this.isAllowedContentEvent( event.getType() ) )
+            {
+                this.handleContentEvent( event );
+            }
+        }
+    }
+
+    private void handleContentEvent( final Event event )
+    {
+        try
+        {
+            final List<Map<String, String>> nodes = (List<Map<String, String>>) event.getData().get( "nodes" );
+
+            nodes.forEach( nodeMap -> {
+                if ( nodeMap.get( "path" ).startsWith( "/content/" ) )
+                {
+                    this.simpleExecutor.execute( () -> handleContentEvent( nodeMap, event.getType() ) );
+                }
+            } );
+        }
+        catch ( Exception e )
+        {
+            LOG.error( "Not able to handle node-event", e );
+        }
+    }
+
+    private boolean isAllowedContentEvent( final String type )
+    {
+        return "node.created".equals( type ) || "node.updated".equals( type ) || "node.pushed".equals( type ) ||
+            "node.renamed".equals( type ) || "node.moved".equals( type ) || "node.deleted".equals( type ) || "node.sorted".equals( type ) ||
+            "node.manualOrderUpdated".equals( type );
+    }
+
+    private void handleContentEvent( final Map<String, String> nodeMap, final String type )
+    {
+        createAdminContext().runWith( () -> {
+
+            final ProjectName currentProjectName = ProjectName.from( RepositoryId.from( nodeMap.get( "repo" ) ) );
+
+            final Project sourceProject = this.projectService.list().
+                stream().
+                filter( project -> currentProjectName.equals( project.getName() ) ).
+                findAny().
+                orElseThrow( () -> new ProjectNotFoundException( currentProjectName ) );
+
+            this.projectService.list().
+                stream().
+                filter( project -> currentProjectName.equals( project.getParent() ) ).
+                forEach( targetProject -> {
+                    final ContentId contentId = ContentId.from( nodeMap.get( "id" ) );
+
+                    final ContentEventsSyncParams.Builder paramsBuilder =
+                        ContentEventsSyncParams.create().contentId( contentId ).sourceProject( sourceProject.getName() ).targetProject(
+                            targetProject.getName() );
+
+                    switch ( type )
```

Comment:
Codacy found an issue: [Switch statements should have a default label](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=6409742)